clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -404,7 +494,16 @@ private void updateGroupSubscription(Set<String> topics) {
 
         log.debug("Performing assignment using strategy {} with subscriptions {}", assignor.name(), subscriptions);
 
-        Map<String, Assignment> assignment = assignor.assign(metadata.fetch(), subscriptions);
+        Map<String, Assignment> assignments = assignor.assign(metadata.fetch(), subscriptions);
+
+        switch (protocol) {
```

Comment:
nit: since we're not doing anything in the EAGER case, couldn't we simplify this:
```java
if (protocol == COOPERATIVE)
  adjustAssignment(ownedPartitions, assignments)
```

Similarly in `onJoinPrepare`

Reply:
Ack here; For `onJoinPrepare` though in the pt.3 this is modified a bit since with EAGER we may still revoke.